### PR TITLE
feat: redirect error page for 401/403

### LINKED ISSUE #74: UI: redirect to an error page when getting 401/3 from the API
The API now returns a meaningful error message with a link to an eIAM page to ask for access to the app. The UI should have a page to show this.

### PR DESCRIPTION
**Pull request by lucafurrer**
When an error 401/403 is raised the user gets routed to a page with a link (provided by the backend) where he can request access to the site.

resolves #74 

Note: New version of Alceaus was needed

### CODE REVIEW

File: ui/src/App.vue [hunk 1]
```diff
@@ -44,6 +45,12 @@ export default Vue.extend({
       this.$store.dispatch('app/dismissMessage', message)
     },
   },
+  errorCaptured (err) {
```
Comment (martinmaillard) [inline ui/src/App.vue:48]: Interesting! I would have never thought of doing it that way. I like it :)

File: ui/src/views/NotAuthorized.vue [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<template>
+  <page-content>
+    <h2 class="title">
+      Missing Authorization
+    </h2>
+    <p>
+      You have not the right to access this site. Please request access it via
+      <a :href="link"> {{ link }}</a>
+    </p>
```
Comment (martinmaillard) [inline ui/src/views/NotAuthorized.vue:9]: We could use the `details` provided by the backend in the error response. But I'm also fine with this.

File: ui/src/App.vue [hunk 3]
```diff
@@ -44,6 +45,12 @@ export default Vue.extend({
...
+    if (err instanceof APIErrorAuthorization) {
+      this.$router.push({ name: 'NotAuthorized', params: { link: err.details.link.href } })
+    }
+    return false
```
Comment (martinmaillard) [inline ui/src/App.vue:52]: Won't that stop any error from propagating?

**Review by martinmaillard — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -37,5 +37,8 @@
     "require": [
       "mocha-setup.js"
     ]
+  },
+  "dependencies": {
+    "alcaeus": "1.0.0-RC.3"
```
Comment (tpluscode) [inline package.json:42]: This must be in `ui/package.json`

**Review by tpluscode — CHANGES_REQUESTED** [review]

File: ui/src/App.vue (hunk 3 above)
Comment (lucafurrer) [reply ui/src/App.vue:52]: True, I missread the documentation.

File: ui/src/views/NotAuthorized.vue (hunk 2 above)
Comment (lucafurrer) [reply ui/src/views/NotAuthorized.vue:9]: I was hesitating about this. The link is read from the error response

File: package.json (hunk 4 above)
Comment (lucafurrer) [reply package.json:42]: Uups

File: ui/src/App.vue (hunk 3 above)
Comment (lucafurrer) [reply ui/src/App.vue:52]: Yes, I misread the documentation, I will fix that.

File: package.json [hunk 5]
```diff
@@ -37,5 +37,6 @@
     "require": [
       "mocha-setup.js"
     ]
-  }
+  },
+  "dependencies": {}
```
Comment (tpluscode) [inline package.json:41]:
```suggestion
  }
```

**Review by vhf — APPROVED** [review]

**Review by tpluscode — APPROVED** [review]
We might wanna work a little on the styling but otherwise it's fine once alcaeus is updated to `1.0.0-RC.4`

**Review by martinmaillard — APPROVED** [review]

**Pull request merged**